go/integration_test/processor/processor_gas_test.go

```diff
@@ -0,0 +1,503 @@
+			},
+		},
+		"SimpleCodeInsufficient": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
+				},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + codePrice - 1,
+				Nonce:     4,
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
+				},
+			},
+			Receipt: tosca.Receipt{
+				Success: false,
+				GasUsed: floria.TxGas + codePrice - 1,
+			},
+			OperaError: fmt.Errorf("gas too low"),
+		},
+		"InputZerosExact": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + floria.TxDataZeroGasEIP2028*10,
+				Nonce:     4,
+				Input:     []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Receipt: tosca.Receipt{
+				Success: true,
+				GasUsed: floria.TxGas + floria.TxDataZeroGasEIP2028*10,
+			},
+		},
+		"InputZerosSufficient": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + floria.TxDataZeroGasEIP2028*10 + excessGas,
+				Nonce:     4,
+				Input:     []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Receipt: tosca.Receipt{
+				Success: true,
+				GasUsed: floria.TxGas + floria.TxDataZeroGasEIP2028*10 + excessGas/10,
+			},
+		},
+		"InputZerosInsufficient": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + floria.TxDataZeroGasEIP2028*10 - 1,
+				Nonce:     4,
+				Input:     []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Receipt: tosca.Receipt{
+				Success: false,
+				GasUsed: 0,
+			},
+			OperaError: fmt.Errorf("gas too low"),
+		},
```

Comment:
Maybe a helper function producing test cases "<name>Exact", "<name>Sufficient", and "<name>Insufficient" for a given scenario would be a nice to have. Wdyt?